public/app/features/alerting/unified/components/rule-editor/LabelsField.tsx

```diff
@@ -25,25 +73,49 @@ const LabelsField: FC<Props> = ({ className }) => {
   const labels = watch('labels');
   const { fields, append, remove } = useFieldArray({ control, name: 'labels' });
 
-  const getDistinctDropdownOptions = (
-    fields: Array<FieldArrayWithId<RuleFormValues, 'labels', 'id'>>,
-    type: AlertLabelDropdownProps['type']
-  ) =>
-    [...new Set(fields.map((field) => field[type]))]
-      .filter((field) => Boolean(field))
-      .map((field) => ({ label: field, value: field }));
+  const labelsByKey = useGetCustomLabels();
+
+  const [selectedKey, setSelectedKey] = useState('');
 
   const keys = useMemo(() => {
```

Comment:
If wonder if it's a good idea to display suggestions on the Test Contact Point modal. This is highly connected to what Alertmanager you use (it might be tricky to get the correct suggestions)

Reply:
Good point. I changed the `LabelsField` component to receive a `suggest` prop that controls whether to display the dropdowns or plain text inputs. 

`suggest=true`
![image](https://user-images.githubusercontent.com/6271380/199594959-cf4d12c6-4551-490e-8eb3-a295c430e4ca.png)

`suggest=false`
![image](https://user-images.githubusercontent.com/6271380/199595080-93c35d05-82f4-4575-a1dd-f1558b005996.png)

Reply:
Nice! Could we re-use code between `LabelsWithoutSuggestions` and `LabelsWithSuggestions`? I see a lot of common code in both components.

Reply:
Totally 👍